src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,727 @@
+                }
+                else
+                    length = semicolonIndex;
+
+                //If there is no linear white space found we treat the entire remaining string as 
+                //parameter value.
+                if (length == -1)
+                    length = s.Length;
+            }
+            else
+            {
+                //if the parameter value starts with a '"' then, we need to look for the
+                //pairing '"' that is not preceded by a "\" ["\" is used to escape the '"']
+                bool found = false;
+                length = startIndex;
+
+                while (!found)
+                {
+                    length = s.IndexOf('"', ++length);
+
+                    if (length == -1)
+                        throw new ArgumentException(SR.InvalidParameterValue);
+
+                    if (s[length - 1] != '\\')
+                    {
+                        found = true;
+                        length++;
+                    }
+                }
+            }
+            return length - startIndex;
+        }
+
+        /// <summary>
+        /// Validating the given token
+        /// The following checks are being made - 
+        /// 1. If all the characters in the token are either ASCII letter or digit.
+        /// 2. If all the characters in the token are either from the remaining allowed cha----ter set.
+        /// </summary>
+        /// <param name="token">string token</param>
+        /// <returns>validated string token</returns>
+        /// <exception cref="ArgumentException">If the token is Empty</exception>
+        private static string ValidateToken(string token)
+        {
+            if (String.CompareOrdinal(token, String.Empty) == 0)
+                throw new ArgumentException(SR.InvalidToken);
+
+            for (int i = 0; i < token.Length; i++)
+            {
+                if (IsAsciiLetterOrDigit(token[i]))
+                {
+                    continue;
+                }
+                else
+                {
+                    if (IsAllowedCharacter(token[i]))
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        throw new ArgumentException(SR.InvalidToken);
+                    }
+                }
+            }
+
+            return token;
+        }
+
+        /// <summary>
+        /// Validating if the value of a parameter is either a valid token or a 
+        /// valid quoted string
+        /// </summary>
+        /// <param name="parameterValue">paramter value string</param>
+        /// <returns>validate parameter value string</returns>
+        /// <exception cref="ArgumentException">If the paramter value is empty</exception>
+        private static string ValidateQuotedStringOrToken(string parameterValue)
+        {
+            if (String.CompareOrdinal(parameterValue, String.Empty) == 0)
```

Comment:
Same question as above.  It seems a bit odd to be explicit about the kind of comparison we want for the characters and to be doing a Compare rather than Equal when what we're checking for is just an empty string.